File: Solutions/Python/plus_one.py
"""
    [EASY]
    66. Plus One

    Concepts:
    - math

    Stats:
        Runtime | 0 ms      [Beats 100%]
        Memory  | 17.87 MB  [Beats 5.23%]
"""

class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        num = 0
        for i, digit in enumerate(digits[::-1]):
            num += digit * (10**i)

        num += 1
        res = []
        while num:
            res.append(num % 10)
            num //= 10

        return res[::-1]

File: Solutions/Cpp/plus_one.cpp
/*
    [EASY]
    66. Plus One

    Concepts:
    - math

    Stats:
        Runtime | 0 ms      [Beats 100%]
        Memory  | 11.60 MB  [Beats 34.92%]
*/

class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        int carry = 1, sum = 0;

        for (int i = digits.size() - 1; i > -1; i--) {
            sum = digits[i] + carry;

            carry = (sum > 9) ? 1 : 0;
            digits[i] = sum % 10;
        }

        if (carry)
            digits.insert(digits.begin(), 1);

        return digits;
    }
};